crates/pallets/starknet/src/lib.rs

```diff
@@ -291,6 +307,115 @@ pub mod pallet {
             Ok(())
         }
 
+		// Submit a Starknet declare transaction.
+        ///
+        /// # Arguments
+        ///
+        /// * `origin` - The origin of the transaction.
+        /// * `transaction` - The Starknet transaction.
+        ///
+        ///  # Returns
+        ///
+        /// * `DispatchResult` - The result of the transaction.
+        ///
+        /// # TODO
+        /// * Compute weight
+        #[pallet::call_index(3)]
+        #[pallet::weight(0)]
+        pub fn add_declare_transaction(_origin: OriginFor<T>, transaction: Transaction) -> DispatchResult {
+            // TODO: add origin check when proxy pallet added
+
+            // Check if contract is deployed
+            ensure!(ContractClassHashes::<T>::contains_key(transaction.sender_address), Error::<T>::AccountNotDeployed);
+
+			// Check class hash is not already declared
+			ensure!(!ContractClasses::<T>::contains_key(transaction.call_entrypoint.class_hash.unwrap()), Error::<T>::ClassHashAlreadyDeclared);
+
+			// Check that contract class is not None
+			ensure!(transaction.contract_class.is_some(), Error::<T>::ContractClassMustBeSpecified);
+
+
+            let block = Self::current_block().unwrap();
+            let state = &mut Self::create_state_reader();
+			let contract_class = transaction.clone().contract_class.unwrap().to_starknet_contract_class();
+			match contract_class {
+				Ok(ref _v) => {
+					log!(info, "Contract class parsed successfully.");
+				}
+				Err(e) => {
+					log!(error, "Contract class parsing failed: {:?}", e);
+					return Err(Error::<T>::InvalidContractClass.into());
+				}
+			}
+
+            match transaction.execute(state, block, TxType::DeclareTx, Some(contract_class.unwrap())) {
```

Comment:
@Same, instead of: 
![Screenshot 2023-03-27 at 09 39 03](https://user-images.githubusercontent.com/45264458/227874160-124413bb-4bdd-46c6-a8a3-958c86f748ec.png)


I would go with:

![Screenshot 2023-03-27 at 09 41 22](https://user-images.githubusercontent.com/45264458/227873884-57087454-813d-48f0-8b3e-b0a10845cd28.png)